snapcraft/__init__.py

```diff
@@ -446,3 +446,8 @@ def str_presenter(dumper, data):
 yaml.add_representer(str, str_presenter)
 yaml.add_representer(OrderedDict, dict_representer)
 yaml.add_constructor(_mapping_tag, dict_constructor)
+
+if common.is_snap():
```

Comment:
this is why the tests fail. So either we drive the logic to set in some other internal module or we
```
from snapcraft.internal.common import is_snap as _is_snap
```
and use it like that.